addon/serializers/json.js

```diff
@@ -843,14 +843,35 @@ var JSONSerializer = Serializer.extend({
   /**
     Check if the given hasMany relationship should be serialized
 
-    @method _shouldSerializeHasMany
+    @method shouldSerializeHasMany
+    @param {DS.Snapshot} snapshot
+    @param {String} key
+    @param {String} relationshipType
+    @return {boolean} true if the hasMany relationship should be serialized
+  */
+  shouldSerializeHasMany(snapshot, key, relationship) {
+    return this.__shouldSerializeHasMany(snapshot, key, relationship);
+  },
+
+  /**
+    Check if the given hasMany relationship should be serialized
+
+    @method __shouldSerializeHasMany
     @private
     @param {DS.Snapshot} snapshot
     @param {String} key
     @param {String} relationshipType
     @return {boolean} true if the hasMany relationship should be serialized
   */
-  _shouldSerializeHasMany(snapshot, key, relationship) {
+  __shouldSerializeHasMany(snapshot, key, relationship) {
```

Comment:
Lets leave `_shouldSerializeHasMany` unchanged in case anyone is calling `this._super()` in `_shouldSerializeHasMany`.